### PR TITLE
LPS-163446: Inconsistency between the REST APIs and the UI using permissions

### PR DESCRIPTION
**Pull request by sergiojimcos**
**What is new**
- Create a new method in `PermissionUtil` that retrieves all the permissions given a `resourceName`.
- Update `base_resource_impl.ftl` to generate `toPermissionsPage` method in all BaseResourceImpls.
- Update all `BaseResourceImpls`

**JIRA Ticket**
https://issues.liferay.com/browse/LPS-163446

### CODE REVIEW

**Review by 4lejandrito — APPROVED** [review]

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/permission/PermissionUtil.java [hunk 1]
```diff
@@ -81,6 +84,57 @@ public static List<ResourcePermission> getResourcePermissions(
...
+		List<Permission> permissions = new ArrayList<>();
+
+		Map<String, Permission> permissionMap = new HashMap<>();
+
+		for (ResourcePermission resourcePermission : resourcePermissions) {
+			Role role = roleLocalService.getRole(
+				resourcePermission.getRoleId());
+
+			if ((roleNames == null) ||
+				((roleNames != null) &&
```
Comment (LuismiBarcos) [inline modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/permission/PermissionUtil.java:110]: I think that the `roleNames != null` check is not necessary as we first check if `roleName == null`

**Review by LuismiBarcos — COMMENTED** [review]
I was reviewing this PR and there is something that could be a problem 😢 

I made some tests and this same to fix the problem for the entries and the lists, I mean, it returns the proper information for both endpoints 
* List `http://localhost:8080/o/headless-delivery/v1.0/sites/{siteId}/blog-postings/permissions'`
* Entries `http://localhost:8080/o/headless-delivery/v1.0/blog-postings/{blogPostingId}permissions`

The proper solution (I know this is a draft, I didn't expect the proper solution here) should be to add this base implementation to the rest builder and generate the code for all APIs. So, where is the problem?

The problem is in the new method added to vulcan-api. Again, we are breaking backward compatibility.

I was checking if we could add this logic to an existing method so we didn't break compatibility and, if a client notices that the endpoints contain a bug, should install an FP/SP/Hotfix, like any other bug.

File: modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/permission/PermissionUtil.java (hunk 1 above)
Comment (4lejandrito) [reply modules/apps/portal-vulcan/portal-vulcan-api/src/main/java/com/liferay/portal/vulcan/permission/PermissionUtil.java:110]: > I was reviewing this PR and there is something that could be a problem 😢
> 
> I made some tests and this same to fix the problem for the entries and the lists, I mean, it returns the proper information for both endpoints
> 
>     * List `http://localhost:8080/o/headless-delivery/v1.0/sites/{siteId}/blog-postings/permissions'`
> 
>     * Entries [http://localhost:8080/o/headless-delivery/v1.0/blog-postings/{blogPostingId}permissions](http://localhost:8080/o/headless-delivery/v1.0/blog-postings/%7BblogPostingId%7Dpermissions)
> 
> 
> The proper solution (I know this is a draft, I didn't expect the proper solution here) should be to add this base implementation to the rest builder and generate the code for all APIs. So, where is the problem?
> 
> The problem is in the new method added to vulcan-api. Again, we are breaking backward compatibility.
> 
> I was checking if we could add this logic to an existing method so we didn't break compatibility and, if a client notices that the endpoints contain a bug, should install an FP/SP/Hotfix, like any other bug.

We discussed the same. Can't we add the method and then backport?

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
Closing this issue because I've seen that there are some inconsistency between two entries

**Pull request closed without merging**